api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/ObservationQuantityCodingTranslatorImplTest.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.when;
+import static org.openmrs.module.fhir2.FhirConstants.UCUM_SYSTEM_URI;
+
+import java.util.Optional;
+
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.SimpleQuantity;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.openmrs.Concept;
+import org.openmrs.ConceptNumeric;
+import org.openmrs.ConceptSource;
+import org.openmrs.module.fhir2.TestFhirSpringConfiguration;
+import org.openmrs.module.fhir2.api.FhirConceptService;
+import org.openmrs.module.fhir2.api.FhirConceptSourceService;
+import org.openmrs.module.fhir2.model.FhirConceptSource;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+
+@ContextConfiguration(classes = TestFhirSpringConfiguration.class, inheritLocations = false)
+public class ObservationQuantityCodingTranslatorImplTest extends BaseModuleContextSensitiveTest {
+	
+	private static final String CONCEPT_UUID = "33fdc8ad-fe4d-499b-93a8-8a991c1d488g";
+	
+	@Mock
+	private FhirConceptService conceptService;
+	
+	@Mock
+	private FhirConceptSourceService conceptSourceService;
+	
+	@Autowired
+	ObservationQuantityCodingTranslatorImpl quantityCodingTranslator;
+	
+	@Before
+	public void setup() {
+		ConceptTranslatorImpl conceptTranslator = new ConceptTranslatorImpl();
+		conceptTranslator.setConceptService(conceptService);
+		conceptTranslator.setConceptSourceService(conceptSourceService);
+		
+		quantityCodingTranslator.setConceptTranslator(conceptTranslator);
+	}
+	
+	@Test
+	public void toFhirResource_shouldTranslateConceptWithCommonUCUMUnitCodes() {
+		ConceptNumeric cn = new ConceptNumeric();
+		
+		cn.setUnits("cm");
```

Comment:
maybe we should extend this to cover all the units defined in https://www.hl7.org/fhir/valueset-ucum-common.html , but i just included some common units to start us off (mostly those called out as important in FM2-64)